mbedtls-platform-support/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) Fortanix, Inc.
+ *
+ * Licensed under the GNU General Public License, version 2 <LICENSE-GPL or
+ * https://www.gnu.org/licenses/gpl-2.0.html> or the Apache License, Version
+ * 2.0 <LICENSE-APACHE or http://www.apache.org/licenses/LICENSE-2.0>, at your
+ * option. This file may not be copied, modified, or distributed except
+ * according to those terms. */
+
+#[cfg(not(feature = "std"))]
+#[allow(unused)]
+#[macro_use]
+extern crate alloc as rust_alloc;
+
+#[cfg(not(feature = "std"))]
+mod alloc_prelude {
+    #![allow(unused)]
+    pub(crate) use rust_alloc::borrow::ToOwned;
+    pub(crate) use rust_alloc::boxed::Box;
+    pub(crate) use rust_alloc::sync::Arc;
+    pub(crate) use rust_alloc::string::String;
+    pub(crate) use rust_alloc::string::ToString;
+    pub(crate) use rust_alloc::vec::Vec;
+    pub(crate) use rust_alloc::borrow::Cow;
+}
+
+#[cfg(any(feature = "spin_threading", feature = "rust_threading", sys_threading_component = "custom"))]
+#[doc(hidden)]
+pub mod threading;
+
+#[cfg(any(feature = "force_aesni_support", target_env = "sgx"))]
+#[doc(hidden)]
+#[no_mangle]
+// needs to be pub for global visibility
+pub extern "C" fn mbedtls_aesni_has_support(_what: u32) -> i32 {
+    return 1;
+}
+
+#[cfg(any(feature = "force_aesni_support", target_env = "sgx"))]
+#[doc(hidden)]
+#[no_mangle]
+// needs to be pub for global visibility
+pub extern "C" fn mbedtls_internal_aes_encrypt(_ctx: *mut mbedtls_sys::types::raw_types::c_void,
+                                                _input: *const u8,
+                                                _output: *mut u8) -> i32 {
+    panic!("AES-NI support is forced but the T-tables code was invoked")
+}
+
+#[cfg(any(feature = "force_aesni_support", target_env = "sgx"))]
+#[doc(hidden)]
+#[no_mangle]
+// needs to be pub for global visibility
+pub extern "C" fn mbedtls_internal_aes_decrypt(_ctx: *mut mbedtls_sys::types::raw_types::c_void,
+                                                _input: *const u8,
+                                                _output: *mut u8) -> i32 {
+    panic!("AES-NI support is forced but the T-tables code was invoked")
+}
+
+
+#[cfg(any(all(feature = "time", feature = "custom_gmtime_r"), sys_time_component = "custom"))]
+#[doc(hidden)]
+#[no_mangle]
+// needs to be pub for global visibility
+pub unsafe extern "C" fn mbedtls_platform_gmtime_r(tt: *const mbedtls_sys::types::time_t, tp: *mut mbedtls_sys::types::tm) -> *mut mbedtls_sys::types::tm {
```

Comment:
Is this implementation compatible with 0.7?

Reply:
Yes.
The function signature actually is same, I just remove the `use` statements.
There is no other import of `time_t` and `tm` 
v0.7 code:
```
#[cfg(all(feature = "time", any(feature = "custom_gmtime_r", feature = "custom_time")))]
use mbedtls_sys::types::time_t;
#[cfg(all(feature = "time", feature = "custom_gmtime_r"))]
use mbedtls_sys::types::tm;

#[cfg(feature="custom_gmtime_r")]
#[doc(hidden)]
#[no_mangle]
pub unsafe extern "C" fn mbedtls_platform_gmtime_r(tt: *const time_t, tp: *mut tm) -> *mut tm {
```